source/particle/world.cc

```diff
@@ -1328,16 +1334,18 @@ namespace aspect
 
           tracer_weight = prm.get_integer("Tracer weight");
 
-          if (prm.get ("Load balancing strategy") == "none")
-            particle_load_balancing = no_balancing;
-          else if (prm.get ("Load balancing strategy") == "remove particles")
-            particle_load_balancing = remove_particles;
-          else if (prm.get ("Load balancing strategy") == "remove and add particles")
-            particle_load_balancing = remove_and_add_particles;
-          else if (prm.get ("Load balancing strategy") == "repartition")
-            particle_load_balancing = repartition;
-          else
-            AssertThrow (false, ExcNotImplemented());
+          const std::vector<std::string> strategies = Utilities::split_string_list(prm.get ("Load balancing strategy"));
+          particle_load_balancing = ParticleLoadBalancing::no_balancing;
+
+          if (std::find(strategies.begin(),strategies.end(), "remove particles") != strategies.end())
+            particle_load_balancing = typename ParticleLoadBalancing::Kind(particle_load_balancing | ParticleLoadBalancing::remove_particles);
+          if (std::find(strategies.begin(),strategies.end(), "add particles") != strategies.end())
```

Comment:
this blocks is dangerous since it does not ensure that we take care of all elements of the list. I think it would be better to write it as a loop over all elements of the list, and for each one test it against all known possibilities via an `if ... else if ...` sequence that ends with an `AssertThrow (false, ...` for the unrecognized element of the list.